package DragonBall;


message Attributes
{
    optional  uint32 hp              = 1; //血量
    optional  uint32 att             = 2; //攻击
    optional  uint32 def             = 3; //防御
    optional  uint32 first           = 4; //先手
    optional  uint32 hit             = 5; //命中
    optional  uint32 dodge           = 6;  //闪避
    optional  uint32 critStrike      = 7; //暴击
    optional  uint32 critRate        = 8; //暴击伤害
    optional  uint32 critResist      = 9; //抗暴击
    optional  double hp_add          = 10; //血量加成
    optional  double att_add         = 11; //攻击加成
    optional  double def_add         = 12; //防御加成
}

message DragonBallList
{
    required   uint32     id          = 1; //龙珠id
    required   Attributes attr        = 2; //龙珠属性
    required   uint32     fight_power = 3; //战力
}


message HeroWithDragonBall
{
    required uint32             heroid       = 1; //英雄id
    repeated DragonBallList     list     = 2; //龙珠
}


message DragonBallInfo
{
    repeated  HeroWithDragonBall dragonBallInfo  = 1;
}